### ISSUE TITLE
Add scale setting for projects

### ISSUE THREAD

**Issue by maks** [democloid/picoTracker#39]
My music theory knowledge isn't great, so would be useful to have a scale setting on projects so that I don't have to keep looking up the notes in a scale that I'm using.

**Comment by democloid** [comment]
How would this work? when scrolling notes it would go through notes of scale rather than every note? Would this be a project level setting or device/global level?

**Comment by maks** [comment]
Yes exactly, when you select a scale, you only get the option to enter notes from that scale. And I would expect it to be per project, so saved in a projects xml data file.

Since its the "closest relative" [this video shows](https://www.youtube.com/watch?v=BeERLyRw5G0) how its implemented on the M8, but really the concept of scale+root key seems almost same on it compared to other grooveboxes I've used like the Electribe 2 and Circuit Tracks and I think its pretty much same on the Elektrons too.

Mind you, the M8 has UI to edit scales, but I only had in mind "hardcoding" a set of scales in a header file for now, as the other grooveboxes I've used only had limited sets of presets and thats been mostly ok for me, plus picotracer is open source, so its really viable for people to contribute scales they want if not already included in the build.
